coalib/bearlib/languages/documentation/DocumentationComment.py

```diff
@@ -33,3 +38,90 @@ def __init__(self, documentation, language,
 
     def __str__(self):
         return self.documentation
+
+    def parse(self):
+        """
+        Parses documentation independent of language and docstyle.
+
+        :return:
+            The list of all the parsed sections of the documentation. Every
+            section is a namedtuple of either ``Description`` or ``Parameter``
+            or ``ReturnValue``.
+        :raises NotImplementedError:
+            When no parsing method is present for the given language and
+            docstyle.
+        """
+        if self.language == "python" and self.docstyle == "default":
+            return self._parse_documentation_with_symbols(
+                (":param ", ": "), ":return: ")
+        else:
+            raise NotImplementedError(
+                "Documentation parsing for {0.language!r} in {0.docstyle!r}"
+                " has not been implemented yet".format(self))
+
+    def _parse_documentation_with_symbols(self, param_identifiers,
+                                          return_identifiers):
+        """
+        Parses documentation based on parameter and return symbols.
+
+        :param param_identifiers:
+            A tuple of two strings with which a parameter starts and ends.
+        :param return_identifiers:
+            The string with which a return description starts.
+        :return:
+            The list of all the parsed sections of the documentation. Every
+            section is a namedtuple of either ``Description`` or ``Parameter``
+            or ``ReturnValue``.
+        """
+        lines = self.documentation.splitlines(keepends=True)
+
+        parse_mode = self.Description
+
+        cur_param = ""
+
+        desc = ""
+        parsed = []
+
+        for line in lines:
+            if line.strip().startswith(param_identifiers[0]):
+                parse_mode = self.Parameter
+                param_offset = line.rfind(
+                    param_identifiers[0]) + len(param_identifiers[0])
+                splitted = line[param_offset:].split(param_identifiers[1], 1)
```

Comment:
`split` could match the above mentioned pattern twice, resulting a doccomment that misses some content. I would use again `find` and start at the last index from `param_offset`.
then you would need to check below for `-1` instead of using the try-except^^